zipline/finance/blotter/blotter.py

```diff
@@ -0,0 +1,188 @@
+#
+# Copyright 2018 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from zipline.finance.cancel_policy import NeverCancel
+from zipline.assets import Asset
+from zipline.utils.input_validation import expect_types
+
+
+class Blotter(object):
```

Comment:
We should make this an ABC (Abstract Base Class). For an example, look at `zipline/data/history_loader.py:HistoryLoader`

Reply:
Done. I've also added @abstractmethod decorators to the unimplemented functions